src/Microsoft.SqlTools.ServiceLayer/ShowPlan/ShowPlanGraph/XmlPlanNodeBuilder.cs

```diff
@@ -45,16 +44,20 @@ public ShowPlanGraph[] Execute(object dataSource)
             {
                 plan = ReadXmlShowPlan(dataSource);
             }
-            
             List<ShowPlanGraph> graphs = new List<ShowPlanGraph>();
 
+            int i = 0;
             foreach (BaseStmtInfoType statement in EnumStatements(plan))
             {
                 // Reset currentNodeId (used through Context) and create new context
                 this.currentNodeId = 0;
                 NodeBuilderContext context = new NodeBuilderContext(new ShowPlanGraph(), this.showPlanType, this);
                 // Parse the statement block
                 XmlPlanParser.Parse(statement, null, null, context);
+                // Get the statement XML for the graph.
+                context.Graph.XmlDocument = GetSingleStatementXml(dataSource, i++);
```

Comment:
incrementing a loop counter in the middle of a loop body is a little unusual code structure.  I would consider (1) putting the increment on its own line to make it more obvious and (2) consider giving it a more descriptive name (e.g. `statementIndex`) that makes it more intuitive why it's being incremented at that point in the code.